### PR TITLE
fix: generic error handling

### PR DESCRIPTION
**Pull request by AlanVerbner**
# Description

Without this change any non handled error would make the server crash as it was not returning a required by schema field.

# Proposed Solution

- Improved error handling code in server
- Added some missing tests to verify everything works as expected

# Important Changes Introduced

n/a

# Testing

`yarn test`

### CODE REVIEW

File: cardano-rosetta-server/test/e2e/server/server.test.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  });
+
+  test('should return a generic error if payload is not valid', async () => {
+    const response = await server.inject({
+      method: 'post',
+      url: '/block',
+      payload: '{ asdasd }'
+    });
+
+    expect(response.statusCode).toEqual(StatusCodes.INTERNAL_SERVER_ERROR);
```
Comment (rhyslbw) [inline cardano-rosetta-server/test/e2e/server/server.test.ts:46]: I would expect a `400` HTTP error code under this condition

File: cardano-rosetta-server/test/e2e/network/network-options-api.test.ts [hunk 2]
```diff
@@ -72,6 +72,11 @@ const allow = {
       message: 'Cant deserialize transaction output from transaction body',
       retriable: false
     },
+    {
+      code: 5000,
+      message: 'An error occurred',
+      retriable: true
```
Comment (rhyslbw) [inline cardano-rosetta-server/test/e2e/network/network-options-api.test.ts:78]: I appreciate this is out of scope, but `retryable` is usually the preferred spelling and is more aesthetically pleasing. It's not really a word, so retry + able

**Review by rhyslbw — CHANGES_REQUESTED** [review]

File: cardano-rosetta-server/test/e2e/network/network-options-api.test.ts (hunk 2 above)
Comment (AlanVerbner) [reply cardano-rosetta-server/test/e2e/network/network-options-api.test.ts:78]: Thanks as I wasn't aware of that :smile: 

That being said, we need to stick to Rosetta spec and it's spelled `retriable` https://www.rosetta-api.org/docs/models/Error.html

File: cardano-rosetta-server/test/e2e/server/server.test.ts (hunk 1 above)
Comment (AlanVerbner) [reply cardano-rosetta-server/test/e2e/server/server.test.ts:46]: As far as we understand, based on `rosetta-sdk-go` [implementation](https://github.com/coinbase/rosetta-sdk-go/blob/master/server/routers.go#L88), it should always return 500

Comment (rhyslbw) [reply cardano-rosetta-server/test/e2e/server/server.test.ts:46]: The spec [confirms](https://www.rosetta-api.org/docs/AccountApi.html#500---error)

File: cardano-rosetta-server/test/e2e/network/network-options-api.test.ts (hunk 2 above)
Comment (rhyslbw) [reply cardano-rosetta-server/test/e2e/network/network-options-api.test.ts:78]: ha, yes of course. That's a beauty!

**Review by rhyslbw — APPROVED** [review]

**Pull request merged**